### PR TITLE
Feature fix simulator

### PR DESCRIPTION
**Pull request by hannesastrom**
Small updates to the cacheray util library. Lots of simulator bugfixes.

The commit messages are bad, but the code is decent.

### CODE REVIEW

**Comment by kimgr** [comment]
Consider breaking separate parts up into smaller prs, e.g. the trace-check thing could probably be broken out. Thanks.

**Comment by hannesastrom** [comment]
There we go

**Comment by kimgr** [comment]
Thanks, I'll take a quick look tomorrow.

File: sim/simulator_java/src/main/java/cache/SimpleCacheFactory.java [hunk 1]
```diff
@@ -13,15 +13,15 @@ public SimpleCacheFactory(){
 	public SimpleCache getCache(Configuration conf){//, SimpleCache nextLevel){
 		SimpleCache cache = null;
 		for(int i = conf.getNumberOfCaches()-1; i >= 0; i--){
-			cache = new OmniCache(conf.getCacheName(i), conf.getCacheSize(i), conf.getBlockSize(i),conf.getAssociativity(i), conf.getWritePolicy(i), conf.getReplacementPolicy(i), cache );
-/*
+			//cache = new OmniCache(conf.getCacheName(i), conf.getCacheSize(i), conf.getBlockSize(i),conf.getAssociativity(i), conf.getWritePolicy(i), conf.getReplacementPolicy(i), cache );
```
Comment (kimgr) [inline sim/simulator_java/src/main/java/cache/SimpleCacheFactory.java:16]: In the future, let's not save commented-out code. There's `git log -p -S` for finding it in history

**Review by kimgr — APPROVED** [review]
The commits are huge, and the commit messages don't really reflect what the purpose of these changes are.

I don't want us to spend a lot of energy splitting things back up now, and I'm not very familiar with the simulator code anyway :). But in the future, please consider smaller chunks, because it's much easier to review and scan through history when searching for a bug. Cheers.

**Pull request merged**